spec/support/components/work_packages/relations.rb

```diff
@@ -116,6 +116,12 @@ def expect_no_new_relation_type(relation_type)
         end
       end
 
+      def expect_no_add_menu_sub_menu
+        within(add_relation_action_menu) do
+          expect(page).to have_no_link("add_relation_sub_menu", wait: 1)
+        end
```

Comment:
This condition is wrong. I tried removing `if should_render_add_relations?` in the code, and it still passes.
Ideally, this kind of negative predicate should use the same selectors as a positive predicate (one which checks for the submenu item presence) so that if the id is changed, the positive predicate will fail, the selector will be updated, which in turn will update the negative predicate as well.

That the problem of checking the absence of something: if the text is renamed, it always passes.

I'll push a fix.